dotcom-rendering/src/amp/lib/get-video-id.ts

```diff
@@ -15,12 +15,15 @@ export const getIdFromUrl = (
 	const url = parse(urlString);
 
 	// Looks for ID in both formats if provided
-	const ids: string[] = [
-		tryQueryParam &&
-			url.query &&
-			new URLSearchParams(url.query).get(tryQueryParam),
-		tryInPath && url.pathname && url.pathname.split('/').pop(),
-	].filter((tryId): tryId is string => !!tryId);
+	const ids: string[] = [];
+	if (tryQueryParam && url.query) {
+		const maybeId = new URLSearchParams(url.query).get(tryQueryParam);
+		if (maybeId) ids.push(maybeId);
+	}
+	if (tryInPath && url.pathname) {
+		const maybeId = url.pathname.split('/').pop();
+		if (maybeId) ids.push(maybeId);
+	}
```

Comment:
I think the immutable version is superior, with a few modifications:

```ts
import { isString } from '@guardian/libs';

const url = new URL(urlString);

const ids: [
	tryQueryParam ? new URLSearchParams(url.query ?? '').get(tryQueryParam) : undefined,
	tryInPath ? url.pathname.split('/').at(-1) : undefined,
].filter(isString);
```

Reply:
Took a bit more modification, personally I still find the mutable version easier to read, but I'll stick with immutability since that's how it worked before!